### PR TITLE
Move ffi/gn to docs/tutorials/gn

### LINKED ISSUE #3006: Reduce the number of checked in generated files
We have quite a few generated files checked in, with CI failing if they are out of date. This happens to me quite a lot, and it's a waste of my time and CI resources (a new commit has to run *everything* again).

Some files very cheap to generate on the fly (`gn-gen` and `diplomat-gen`), while some like testdata are not. Can we remove the cheap files?

### PR DESCRIPTION
**Pull request by sffc**
Fixes #3006

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

File: .gn
[the remark this answers is not in the document]
Comment (sffc) [reply .gn:1]: It defines the root, which must be an ancestor of code being built

File: docs/tutorials/gn/BUILD.gn [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 group("default") {
   deps = [
-    "//ffi/gn/hello_world:print_hello",
-    "//ffi/gn/icu4x:icu_capi"
+    "//docs/tutorials/gn/hello_world:print_hello",
```
[the remark this answers is not in the document]
Comment (sffc) [reply docs/tutorials/gn/BUILD.gn:7]: Either this or using a symlink; prefer making the GN root the repo root. Clients vendoring ICU4X don't need this part of the setup

File: docs/tutorials/gn/Cargo.toml [hunk 2]
```diff
@@ -10,9 +10,9 @@ edition = "2021"
 
 [dependencies]
 # Note: Only dependencies that are explicitly listed here are given public GN build targets.
-icu_capi = { version = "1.0", path = "../../ffi/capi", default-features = false, features = ["default_components"] }
-icu = { version = "1.0", path = "../../components/icu", default-features = false }
-icu_provider = { version = "1.0", path = "../../provider/core" }
+icu_capi = { version = "1.0", default-features = false, features = ["default_components"] }
```
[the remark this answers is not in the document]
Comment (sffc) [reply docs/tutorials/gn/Cargo.toml:13]: Yes

File: tools/make/gn.toml
[the remark this answers is not in the document]
Comment (sffc) [reply tools/make/gn.toml:1]: #4546

File: Cargo.toml [hunk 3]
```diff
@@ -103,7 +103,7 @@ members = [
 # implicitly excluded from the main workspace.
 exclude = [
     # GNaw needs to be run from outside the workspace for features to work
-    "ffi/gn",
+    "docs/tutorials/gn",
```
[the remark this answers is not in the document]
Comment (sffc) [reply Cargo.toml:106]: #4546

**Pull request merged**